accepted/future-releases/abstract-external-fields/feature-specification.md

```diff
@@ -0,0 +1,240 @@
+client code that declares getter/setter pairs with a non-trivial amount of
+redundancy: The name is specified twice, the type is specified twice, and
+there might be a need for two copies of metadata specifying the native
+representation. So `dart:ffi` uses regular instance variable declarations
+today.
+
+Similarly, a class can of course declare an abstract getter/setter pair if
+needed, but this again causes the name and type to be specified twice.
+
+As a workaround, it is possible to use a concrete instance variable as a
+replacement for an abstract getter and possibly setter, by simply ignoring
+the implementation. The concrete variable declaration will add the
+implicitly induced getter and possible setter to the interface of the
+enclosing class (let us call it `C`). Any concrete subtypes (`class D
+implements C ..`) will be required to implement said accessors. So far, it
+just works.
+
+However, if `D` is a _subclass_ of `C` then `D` is not required to
+implement the accessors. Instead, `D` tacity inherits the implementation of
+the concrete variable. This may be a bug, because `C` is designed for having
+these accessors overridden to do specific things that the concrete variable
+will not do. Also, if `D` _does_ override the accessors with a getter and
+setter as intended, the storage reserved for the concrete instance variable
+will be a space leak (it is unused).
+
+Finally, if null-safety is enabled then a concrete instance variable
+declaration (say, `int foo;`) is a compile-time error if the type of the
+variable is non-nullable, unless the variable is initialized in the
+initializer list of each generative constructor of the class. This problem
+arises both in connection with `dart:ffi` and for a an instance variable
+which is used as a replacement for an abstract getter and possibly setter.
+
+So the existing workaround of using a concrete instance variable to emulate
+an abstract instance variable is inconvenient, error prone, and confusing
+for a reader of the code, and for a writer of a subclass, and similar
+problems exist when a regular variable is used as a replacement for an
+external getter and possibly setter.
+
+
+## Design Idea
+
+In response to the issues above, this document introduces abstract and
+external variables.
+
+The basic idea is that an abstract variable is syntactic sugar for an
+abstract getter and possibly an abstract setter, and similarly for an
+external variable, in both cases such that there is no storage and no
+implementation of the getter and possible setter, and they have the same
+signatures as the ones that would be induced implicitly by a concrete
+variable declaration.
+
+The next section is the normative text that specifies the syntax and
+semantics that realize this idea.
+
+
+## Feature Specification
+
+An _abstract instance variable declaration_ is an instance variable
+declaration prefixed by the modifier `abstract`. It must not be late, and
```

Comment:
It --> The declaration

Be careful about using an unqualified "it". The previous sentence contains two noun phrases, and it's not obvious that "it" binds to the latter.

Reply:
Done in new PR.